### ISSUE TITLE
Firebird 4 embedded doesn't allow creating/connecting database with non ANSI characters in path

### ISSUE THREAD

**Issue by Rand-Random** [FirebirdSQL/firebird#7101]
See the following issue for further information:
https://github.com/FirebirdSQL/NETProvider/issues/1019

And the comment from @cincuranet in https://github.com/FirebirdSQL/NETProvider/issues/1019#issuecomment-1013135494
> For the filesystem operations it is still using ANSI (at least was using in FB3). Maybe it can be changed in FB5.

**Comment by aafemt** [comment]
An attempt to create an Unicode Firebird https://github.com/aafemt/firebird/tree/unicode shown that it is not feasible.

**Comment by asfernandes** [comment]
Can't this be easily done with this?

https://docs.microsoft.com/en-us/windows/apps/design/globalizing/use-utf8-code-page

**Comment by Rand-Random** [comment]
Based on the comment
https://github.com/FirebirdSQL/NETProvider/issues/1019#issuecomment-1013146029

and the test I did here
https://github.com/FirebirdSQL/NETProvider/issues/1019#issuecomment-1013222851

It seems it is an error/oversight in the .net provider, and that Firebird requires the flag `isc_dpb_utf8_filename` if the filename is passed as UTF8 encoded, but @cincuranet would need to verfiy it, I can only state what I am observing and not exactly "know" where the issue is.

**Comment by aafemt** [comment]
No, this flag won't help because is intended only to handle characters that are presented in server ANSI codepage but not client one.

**Comment by Rand-Random** [comment]
But, I did test it and it did help.

This new issue
https://github.com/FirebirdSQL/NETProvider/issues/1020
seems to describe my problem and the things I am observing, just more technical, things I simply don't know.

I am closing this issue for now.

**Comment by aafemt** [comment]
It looks like you mixed up non-ASCII and non-ANSI characters. Besides, the link describes completely unrelated problem.

**Comment by asfernandes** [comment]
> An attempt to create an Unicode Firebird https://github.com/aafemt/firebird/tree/unicode shown that it is not feasible.

IMO its much simple and better to use strings/PathNames interpreted as UTF8 and convert them to wide just before call W-functions.

**Comment by aafemt** [comment]
It was the simplest part of work. The real problems snowballed after that.

**Comment by asfernandes** [comment]
> It was the simplest part of work. The real problems snowballed after that.

For example?

**Comment by aafemt** [comment]
isql and gstat to correctly show the database name including work via services API.
File name correctly shown in error message about failed connection.
Firebird being run from path containing non-ANSI characters. That time ICU wasn't able to be loaded from such path at all.
